src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -1013,6 +1013,11 @@ object FunctionRegistry {
 
   registerMethod("split", (s: String, p: String, n: Int) => s.split(p, n): IndexedSeq[String])
 
+  registerMethod("lower", (s: String) => s.toLowerCase)
+  registerMethod("upper", (s: String) => s.toUpperCase)
+  registerMethod("strip", (s: String) => s.trim())
+  registerMethod("contains", (s: String, t: String) => s.contains(t))
```

Comment:
I want to be difficult: no new FunctionRegistry methods without corresponding IR methods.

Reply:
can you point me to the code to mimic?

Reply:
in is.hail.expr.ir.functions. You probably want a StringFunctions to mimic MathFunctions, and then you need to call registerAll in IRFunctionRegistry.